cluster/gce/gci/configure-helper_test.go

```diff
@@ -0,0 +1,215 @@
+	kmsPluginContainerIndex        = 0
+	apiServerContainerIndexNoKMS   = 0
+	apiServerContainerIndexWithKMS = 1
+
+	//	command": [
+	//   "/bin/sh", - Index 0
+	//   "-c",      - Index 1
+	//   "exec /usr/local/bin/kube-apiserver " - Index 2
+	execArgsIndex = 2
+
+	socketVolumeMountIndexKMSPlugin = 1
+	socketVolumeMountIndexApiServer = 0
+)
+
+type kubeApiServerEnv struct {
+	KubeHome                     string
+	EncryptionProviderConfig     string
+	EncryptionProviderConfigPath string
+	ETCDKMSKeyID                 string
+}
+
+type kubeApiServerManifestTestCase struct {
+	*ManifestTestCase
+	apiServerContainer v1.Container
+	kmsPluginContainer v1.Container
+}
+
+func NewKubeApiServerManifestTestCase(t *testing.T) *kubeApiServerManifestTestCase {
+	c := &kubeApiServerManifestTestCase{}
+	c.ManifestTestCase = NewManifestTestCase(kubeApiServerManifestFileName, kubeApiServerStartFuncName, []string{kmsPluginManifestFileName}, t)
+
+	return c
+}
+
+func (c *kubeApiServerManifestTestCase) getContainers() {
+	if len(c.pod.Spec.Containers) == 0 {
+		c.t.Fatal("Expected at least one container in apiserver pod, but got 0.")
+	}
+
+	if len(c.pod.Spec.Containers) > 2 {
+		c.t.Fatalf("Expected at most two containers in apiserver pod, but got %d.", len(c.pod.Spec.Containers))
+	}
+
+	if len(c.pod.Spec.Containers) == 2 {
+		c.apiServerContainer = c.pod.Spec.Containers[apiServerContainerIndexWithKMS]
+		c.kmsPluginContainer = c.pod.Spec.Containers[kmsPluginContainerIndex]
+	} else {
+		c.apiServerContainer = c.pod.Spec.Containers[apiServerContainerIndexNoKMS]
+	}
+}
+
+func (c *kubeApiServerManifestTestCase) invokeTest(e kubeApiServerEnv) {
+	c.MustInvokeFunc(deployHelperEnv, e)
+	c.MustLoadPodFromManifest()
+	c.getContainers()
+}
+
+func TestEncryptionProviderFlag(t *testing.T) {
+	c := NewKubeApiServerManifestTestCase(t)
+	defer c.TearDown()
+
+	var e = kubeApiServerEnv{
+		KubeHome:                     c.kubeHome,
+		EncryptionProviderConfig:     base64.StdEncoding.EncodeToString([]byte("FOO")),
+		EncryptionProviderConfigPath: filepath.Join(c.kubeHome, "encryption-provider-config.yaml"),
+	}
+
+	c.invokeTest(e)
+
+	expectedFlag := fmt.Sprintf("--experimental-encryption-provider-config=%s", e.EncryptionProviderConfigPath)
+	execArgs := c.apiServerContainer.Command[execArgsIndex]
+	if !strings.Contains(execArgs, expectedFlag) {
+		c.t.Fatalf("Expected %s flag, but did not find it in %s", expectedFlag, execArgs)
+	}
+}
+
+func TestEncryptionProviderConfig(t *testing.T) {
+	c := NewKubeApiServerManifestTestCase(t)
+	defer c.TearDown()
+
+	p := filepath.Join(c.kubeHome, "encryption-provider-config.yaml")
+	var e = kubeApiServerEnv{
+		KubeHome:                     c.kubeHome,
+		EncryptionProviderConfig:     base64.StdEncoding.EncodeToString([]byte("FOO")),
+		EncryptionProviderConfigPath: p,
+	}
+
+	c.MustInvokeFunc(deployHelperEnv, e)
+
+	if _, err := os.Stat(p); os.IsNotExist(err) {
+		c.t.Fatalf("Expected encryption provider config to be written to %s", p)
```

Comment:
simply check err != nil, IsNotExist falls under it but you also catch any other issues (e.g. permissions)